lib/rules/media-feature-name-unit-allowed-list/index.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const { TokenType } = require('@csstools/css-tokenizer');
+const { isTokenNode } = require('@csstools/css-parser-algorithms');
+const {
+	isMediaFeaturePlain,
+	isMediaFeatureRange,
+	parse,
+} = require('@csstools/media-query-list-parser');
+const validateObjectWithArrayProps = require('../../utils/validateObjectWithArrayProps');
+const { isString } = require('../../utils/validateTypes');
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const flattenArray = require('../../utils/flattenArray');
+const atRuleParamIndex = require('../../utils/atRuleParamIndex');
+
+const ruleName = 'media-feature-name-unit-allowed-list';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (unit, name) => `Unexpected unit "${unit}" for name "${name}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/media-feature-name-unit-allowed-list',
+};
+
+/** @type {import('stylelint').Rule} */
```

Comment:
[trivial] We can make the code a bit type-safe:

```suggestion
/** @type {import('stylelint').Rule<Record<string, string | string[]>>} */
```